src/azure-cli/azure/cli/command_modules/vm/_template_builder.py

```diff
@@ -1446,6 +1447,12 @@ def build_vmss_resource(cmd, name, computer_name_prefix, location, tags, overpro
             }
         }
 
+    if security_posture_reference_id or security_posture_reference_exclude_extensions:
+        virtual_machine_profile['securityPostureReference'] = {
+            'id': security_posture_reference_id,
+            'excludeExtensions': security_posture_reference_exclude_extensions
+        }
```

Comment:
If one attribute is None and the other attribute is not None, then we will pass one attribute as None to the server. Is this expected?